app/audit/AuditDetails.scala

```diff
@@ -16,56 +16,58 @@
 
 package audit
 
-import cats.data.Ior
-import config.Constants
+import models.BoxId
 import models.ChannelType
-import models.EORINumber
-import models.TURN
+import play.api.libs.functional.syntax.toFunctionalBuilderOps
+import play.api.libs.functional.syntax.unlift
 import play.api.libs.json.JsObject
-import play.api.libs.json.Json
 import play.api.libs.json.OWrites
+import play.api.libs.json.__
 
 sealed abstract class AuditDetails {
   def channel: ChannelType
   def customerId: String
   def json: JsObject
-
-  lazy val writeAuditDetails: JsObject =
-    Json.obj(
-      "channel"    -> channel,
-      "customerId" -> customerId
-    ) ++ json
 }
 
-case class AuthenticatedAuditDetails(channel: ChannelType, enrolmentId: Ior[TURN, EORINumber], json: JsObject) extends AuditDetails {
-
-  lazy val customerId: String =
-    enrolmentId.fold(
-      turn => turn.value,
-      eoriNumber => eoriNumber.value,
-      (_, eoriNumber) => eoriNumber.value
-    )
-
-  lazy val enrolmentType: String =
-    enrolmentId.fold(
-      _ => Constants.LegacyEnrolmentKey,
-      _ => Constants.NewEnrolmentKey,
-      (_, _) => Constants.NewEnrolmentKey
-    )
-
-  lazy val writeAuthenticatedAuditDetails: JsObject =
-    writeAuditDetails ++
-      Json.obj("enrolmentType" -> enrolmentType)
-}
+case class AuthenticatedAuditDetails(channel: ChannelType, customerId: String, enrolmentType: String, json: JsObject) extends AuditDetails
 
 object AuthenticatedAuditDetails {
 
-  implicit val writes: OWrites[AuthenticatedAuditDetails] = _.writeAuthenticatedAuditDetails
+  implicit val writes: OWrites[AuthenticatedAuditDetails] = (
+    (__ \ "channel").write[ChannelType] and
+      (__ \ "customerId").write[String] and
+      (__ \ "enrolmentType").write[String] and
+      (__ \ "json").write[JsObject]
+  )(unlift(AuthenticatedAuditDetails.unapply))
 }
 
 case class UnauthenticatedAuditDetails(channel: ChannelType, customerId: String, json: JsObject) extends AuditDetails
 
 object UnauthenticatedAuditDetails {
 
-  implicit val writes: OWrites[UnauthenticatedAuditDetails] = _.writeAuditDetails
+  implicit val writes: OWrites[UnauthenticatedAuditDetails] = (
+    (__ \ "channel").write[ChannelType] and
```

Comment:
We seem to have a few repeated strings in the code, is it worth adding a file for constants if we don't already have one?

Reply:
More my opinion here, but in this scenario, no. For example - if this "channel" changed here, it is only because this specific model has had a field name change. Thus, adding it as a constant will add no value as there is actually no link between all the places we specify it - and if we change one field name, while we might change all the models with the same field name, we aren't necessarily going to do that.

If the string was a value that, if it changed, it's likely it should be changed in all other places its specified, then it should be a constant - that's not the case here.

Additionally here, it's far easier to see the link between the case class field name and the string.